core/domain/exp_services.py

```diff
@@ -822,6 +823,11 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
 
     exploration.version += 1
 
+    # Trigger statistics model update.
+    taskqueue_services.defer(
+        stats_services.handle_stats_creation,
+        taskqueue_services.QUEUE_NAME_DEFAULT, exploration, change_list)
```

Comment:
I'm not entirely certain that you can pass domain objects through the taskqueue. Have you checked that they actually do come out as well-formed domain objects at the other end? (If so, how did you verify this?) If it doesn't work, you might need to do pre-computation here or pass just the needed bits.

Reply:
@seanlip There seems to be no issues there. Check lines 76-83 of stats_services_test file - I wouldn't have been able to retrieve the ExplorationStatsModel there if the domain object hadn't been passed through the task queue.

Reply:
Removed async calls - no issue here